File: src/main/java/ordilov/randomplay/playlist/domain/PlaylistCommand.java
package ordilov.randomplay.playlist.domain;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import ordilov.randomplay.member.domain.Member;
import ordilov.randomplay.playlist.interfaces.PlaylistDto.AddPlaylistRequest;
import ordilov.randomplay.playlist.interfaces.PlaylistDto.AddVideoRequest;

@Getter
public class PlaylistCommand {

  @Getter
  @RequiredArgsConstructor
  public static class PlaylistCreateRequest{
    private final Long memberId;
    private final String title;
    public Playlist toEntity(Member member) {
      return Playlist.builder()
          .member(member)
          .title(title)
          .build();
    }
  }




  @Getter
  public static class YoutubeListRequest {

    private final Long memberId;
    private final Long playlistId;
    private final String youtubePlaylistId;

    public YoutubeListRequest(Long id, AddPlaylistRequest request) {
      this.memberId = id;
      this.playlistId = request.getPlaylistId();
      this.youtubePlaylistId = request.getYoutubePlaylistId();
    }
  }

  @Getter
  public static class YoutubeVideoRequest {

    private final Long memberId;
    private final Long playlistId;
    private final String url;

    public YoutubeVideoRequest(Long id,
        Long playlistId,
        AddVideoRequest request) {
      this.memberId = id;
      this.playlistId = playlistId;
      this.url = request.getUrl();
    }
  }

  @Getter
  @RequiredArgsConstructor
  public static class PlaylistUpdateRequest {

    private final Long memberId;
    private final Long playlistId;
    private final String title;

  }

  @Getter
  @RequiredArgsConstructor
  public static class PlaylistDeleteRequest {

    private final Long memberId;
    private final Long playlistId;

  }

  @Getter
  @RequiredArgsConstructor
  public static class PlaylistItemDeleteRequest {

    private final Long memberId;
    private final Long playlistId;
    private final Long playlistItemId;
  }
}

File: src/main/java/ordilov/lissn/playlist/application/port/in/PlaylistCommand.kt
package ordilov.lissn.playlist.application.port.`in`

import AddPlaylistRequest
import AddVideoRequest

data class PlaylistCreateRequest(
    val memberId: Long,
    val title: String
)

data class YoutubeListRequest(
    val memberId: Long,
    val playlistId: Long,
    val youtubePlaylistId: String,
) {

    constructor(id: Long, request: AddPlaylistRequest) : this(
        memberId = id,
        playlistId = request.playlistId,
        youtubePlaylistId = request.youtubePlaylistId
    )
}

data class YoutubeVideoRequest(
    val memberId: Long,
    val playlistId: Long,
    val url: String
) {

    constructor (
        id: Long,
        playlistId: Long,
        request: AddVideoRequest
    ) : this(
        memberId = id,
        playlistId = playlistId,
        url = request.url
    )
}

data class PlaylistUpdateRequest(
    val memberId: Long,
    val playlistId: Long,
    val title: String
)

data class PlaylistDeleteRequest(
    val memberId: Long,
    val playlistId: Long
)

data class PlaylistItemDeleteRequest(
    val memberId: Long,
    val playlistId: Long,
    val playlistItemId: Long
)